client/src/main/scala/com.gu.contentapi.client/utils/CapiModelEnrichment.scala

```diff
@@ -51,12 +58,96 @@ object CapiModelEnrichment {
         tagExistsWithId("tone/analysis") -> Analysis,
         isComment -> Comment,
         tagExistsWithId("tone/features") -> Feature,
-        liveBlog -> Live,
-        deadBlog -> Article
+        isLiveBlog -> Live,
+        isDeadBlog -> Article
       )
 
-      val result = predicates.collectFirst { case (predicate, design) if predicate(content) => design }
+      val result = getFromPredicate(content, predicates)
       result.getOrElse(defaultDesignType)
     }
   }
+
+  implicit class RenderingFormat(val content: Content) extends AnyVal {
+
+    def design: Design = {
+
+      val defaultDesign: Design = ArticleDesign
+
+      val predicates: List[(ContentFilter, Design)] = List(
+        tagExistsWithId("artanddesign/series/guardian-print-shop") -> PrintShopDesign,
+        tagExistsWithId("tone/matchreports") -> MatchReportDesign,
+        isMedia -> MediaDesign,
+        isReview -> ReviewDesign,
+        tagExistsWithId("tone/analysis") -> AnalysisDesign,
+        isComment -> CommentDesign,
+        tagExistsWithId("tone/features") -> FeatureDesign,
+        tagExistsWithId("tone/recipes") -> RecipeDesign,
+        tagExistsWithId("tone/matchreports") -> MatchReportDesign,
+        tagExistsWithId("tone/interview") -> InterviewDesign,
+        tagExistsWithId("tone/editorials") -> EditorialDesign,
+        tagExistsWithId("tone/quizzes") -> QuizDesign,
+        isPhotoEssay -> PhotoEssayDesign,
+        isLiveBlog -> LiveBlogDesign,
+        isDeadBlog -> DeadBlogDesign
+      )
+
+      val result = getFromPredicate(content, predicates)
+      result.getOrElse(defaultDesign)
+    }
+
+    def theme: Theme = {
+      val defaultTheme: Theme = NewsPillar
+
+      val specialReportTags: Set[String] = Set(
+        "business/series/undercover-in-the-chicken-industry",
+        "business/series/britains-debt-timebomb",
+        "world/series/this-is-europe",
+        "environment/series/the-polluters",
+        "news/series/hsbc-files",
+        "news/series/panama-papers",
+        "us-news/homan-square",
+        "uk-news/series/the-new-world-of-work",
+        "world/series/the-new-arrivals",
+        "news/series/nauru-files",
+        "us-news/series/counted-us-police-killings",
+        "australia-news/series/healthcare-in-detention",
+        "society/series/this-is-the-nhs"
+      )
+
+      def isPillar(pillar: String): ContentFilter = c => c.pillarName.contains(pillar)
+
+      val isSpecialReport: ContentFilter = c => c.tags.exists(t => specialReportTags(t.id))
+      val isOpinion: ContentFilter = c => isComment(c) && isPillar("News")(c) || isPillar("Opinion")(c)
```

Comment:
> I was having this discussion with @nicl and @gtrufitt earlier, is it better to use a more descriptive name instead of 'c'? I'm actually confused here what 'c' represents! 

Happy to change this. I was following the convention from the existing code, but I agree that we can make it simpler. `c` is something of the `Content` type, and overall this `val` is a `ContentFilter` which is a function mapping from `Content` to a `Boolean`.

> The isComment(c) but might be a problem in that we need to both check designtype and pillar in combination here
I can maybe make this clearer in the code but this is actually doing that. `isComment(c)` checks if it's of design type comment, and then `isPillar("News")(c)` checks the pillar.

Reply:
I suspect it's more that I just can't read Scala that well but I can imagine this code being something that is read a lot in future by people trying to understand how the model works at the Guardian and a lot of them won't be Scala devs either so the clearer the better I guess.

Reply:
Another good place to explain to people how a pice of code works is with the tests. It might be better to focus there and let the code be what it needs to be?

Reply:
An alternative to changing the variable name would be to add a type annotation. So:

```scala
def isPillar(pillar: String): ContentFilter = (c: Content) => c.pillarName.contains(pillar)
```